starter-code/src/viking.js

```diff
@@ -1,11 +1,39 @@
-// Soldier
-function Soldier() {}
+ // Soldier
+ function Soldier(healthArg, strengthArg) {
+  this.health = healthArg;
+  this.strength = strengthArg;
+}
+Soldier.prototype.attack = function (){
+return this.strength;
+}
+Soldier.prototype.receiveDamage = function (damage){
+this.health -=  damage;
+};
+
 
-// Viking
-function Viking() {}
 
-// Saxon
-function Saxon() {}
 
-// War
-function War() {}
+// Viking
+
+function Viking(name,healthArg, strengthArg) {
+  Soldier.call(this, healthArg, strengthArg);
+  this.name = name;
+  console.log(this.name);
+  }
+  Viking.prototype = Object.create(Soldier.prototype);
+  Viking.prototype.constructor = Viking;
+  //var viking = new Viking (name, healthArg, strengthArg);
+  
+  var viking = new Viking (name, health, strength);
```

Comment:
do not need to create this. The vikings and Saxons are created in the Jasmine tests.